libvast/src/command.cpp

```diff
@@ -357,6 +357,34 @@ caf::error parse_impl(invocation& result, const command& cmd,
   return parse_impl(result, **i, position + 1, last, target);
 }
 
+void fixup_options(invocation& inv) {
+  using namespace std::string_literals;
+  auto move_option_arg = [&](auto cmd, auto format, auto opt) {
+    auto path = "vast."s + cmd + '.' + format + '.' + opt;
+    if (auto x = caf::get_if<std::string>(&inv.options, path)) {
+      // The logger isn't initialized yet.
+#if VAST_LOG_LEVEL >= VAST_LOG_LEVEL_WARNING
+      fmt::print(stderr,
+                 "The option '{}' is deprected for the '{}' subcommand and "
+                 "should be specified at the {} subcommand instead\n",
+                 opt, format, cmd);
+#endif
+      put(inv.options, "vast."s + cmd + '.' + opt, *x);
+    }
+  };
+  for (const auto& cmd : {"import", "spawn.source"})
+    for (const auto& format :
+         {"csv", "json", "suricata", "syslog", "zeek", "zeek-json"})
+      for (const auto& opt :
+           {"listen", "read", "schema", "schema-file", "type", "uds"})
+        move_option_arg(cmd, format, opt);
+  for (const auto& cmd : {"export", "spawn.sink"})
+    for (const auto& format :
+         {"arrow", "ascii", "csv", "json", "null", "pcap", "zeek"})
```

Comment:
I don't like that we must list this manually here. Is there away we can refactor this so it uses the existing command structure? If there isn't, then we should at least add comments to application.cpp with a hint to check this code as well.

Reply:
I really don't like the idea of adding extra machinery to the command API to specify these relocations. It would require modifications to the options parser to do it properly.

Reply:
Then please add comments to application.cpp.